core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -4226,56 +4232,107 @@ wasm_interp_call_wasm(WASMModuleInstance *module_inst, WASMExecEnv *exec_env,
         }
     }
     else {
+#if WASM_ENABLE_FAST_JIT != 0 || WASM_ENABLE_JIT != 0
+        /*  priority of choosing running mode:
+         * 1. user-set module instance running mode
+         * 2. user-set default running mode
+         * 3. system default running mode */
+        RunningMode running_mode;
+        running_mode =
+            wasm_runtime_get_running_mode((wasm_module_inst_t)module_inst);
+
+        if (running_mode == 0)
+            running_mode = wasm_runtime_get_default_running_mode();
+
+        if (running_mode == 0) {
+            /* if user didn't set one
+             * use the system default running mode:
+             * With only interpreter and one of JIT enabled run LLVM JIT/Fast
+             * JIT. With interpreter and both JIT enabled, run multi-tier JIT */
+#if WASM_ENABLE_JIT != 0 && WASM_ENABLE_FAST_JIT != 0 \
+    && WASM_ENABLE_LAZY_JIT != 0
+            running_mode = Mode_Multi_Tier_JIT;
+#elif WASM_ENABLE_JIT != 0
+            running_mode = Mode_LLVM_JIT;
+#elif WASM_ENABLE_FAST_JIT != 0
+            running_mode = Mode_Fast_JIT;
+#endif
+        }
+#endif
```

Comment:
running mode should have been resolved and set in wasm_instantiate when instantiating or wasm_runtime_set_running_mode, no need to resolve it again.